npm/react/src/createMount.ts

```diff
@@ -109,14 +95,14 @@ export const makeMountFn = (
     }
 
     return (
-      // Separate alias and returned value. Alias returns the component only, and the thenable returns the additional functions
       cy.wrap<React.ReactNode>(userComponent, { log: false })
-      .as(displayName)
       .then(() => {
         return cy.wrap<MountReturn>({
           component: userComponent,
           rerender: (newComponent) => makeMountFn('rerender', newComponent, options, key, internalMountOptions),
-          unmount: internalMountOptions.unmount,
+          unmount: () => {
+            throw new Error('unmount is no longer supported. See https://docs.cypress.io/guides/references/migration-guide#Component-Testing-Changes to migrate.')
```

Comment:
Let's format this error based on our discussion with Brian (docsUrl, onLink, open-mode hyperlink vs run-mode)